Add versioning using resource file

anjannath opened the pull request:
use go-winres to set the version for executable

go-winres uses win resource file to set the version of the executable
it uses the version from the winres/winres.json file, from it creates
a .syso file which is automatically picked by go build

more information about go-winres can be found at:
https://github.com/tc-hib/go-winres/tree/main#usage

anjannath committed:
use go-winres to set the version for executable

go-winres uses win resource file to set the version of the executable
it creates a .syso file which is automatically picked by go build

we use the 'go-winres.exe simple' command and pass the version and product
name for the executable using flags

more information about go-winres can be found at:
https://github.com/tc-hib/go-winres/tree/main#usage

anjannath committed:
add Makefile target to vendor dependencies

run vendor target to vendor the dependencies of go-winres
this'll be needed when building in a disconnected env

anjannath committed:
add .gitignore file

cfergeau commented on Makefile:
```diff
@@ -9,3 +9,9 @@ include tools/tools.mk
+vendor:
+	go mod tidy
+	go mod vendor
+	cd $(TOOLS_DIR) && go mod tidy && go mod vendor
```
Maybe this could be a `vendor-tools` target in `tools/tools.mk` to keep everything self-contained?

cfergeau commented on winres/winres.json:
```diff
@@ -0,0 +1,28 @@
+            "PrivateBuild": "",
+            "ProductName": "",
+            "ProductVersion": "",
+            "SpecialBuild": ""
```
Do we have to set all these empty fields? If we omit them, will they get "" as their default value?
Might be worth setting `CompanyName` and `ProductName`?
Or make use of `go-winres simply` which is `a simpler make that does not rely on a json file.` ?

cfergeau replied:
> Do we have to set all these empty fields?

I mean, do we have to explicit list them in the json file if their value is ""? If we omit them, will they get set to "" as a default?

anjannath replied:
Yes these have to explicitly listed, initially it created more json objects, i tried removing them one by one and this whole object is the bare-minimum needed, otherwise the version is not set

cfergeau commented on winres/winres.json:
```diff
@@ -0,0 +1,28 @@
+    "#1": {
+      "0000": {
+        "fixed": {
+          "file_version": "0.0.0.1",
```
Do we need to update this to set the version?

anjannath replied:
yes, it'd be nice to get rid of  the json file, will try out the suggestion from https://github.com/crc-org/win32-background-launcher/pull/1#discussion_r1377504359

cfergeau commented on main.go:
```diff
@@ -1,3 +1,5 @@
+//go:generate go-winres.exe make
```
Have you considered using `//go:generate go-winres make --product-version=git-tag`  to build the version directly from the latest git tag?

anjannath replied:
updated the PR to use this and removed the json file

anjannath replied:
had to remove this, and added the versioning in the `Makefile` since we'll have to upload the source tarball and that doesn't have git history it fails while building on brew, i have updated  https://github.com/crc-org/win32-background-launcher/pull/1/commits/3be44d1ac35d7410dddb702cf55b31d8479cd8f8 in the PR again!

anjannath committed:
update module name to github.com/crc-org/win32-background-launcher

cfergeau commented on tools/go.mod:
```diff
@@ -0,0 +1,14 @@
+module github.com/cfergeau/win32-background-launcher
```
This should be `github.com/crc-org/win32-background-launcher/tools` (and I've seen now that cfergeau -> crc-org is addressed in one of the next commits)

cfergeau commented on tools/go.mod:
```diff
@@ -1,4 +1,4 @@
-module github.com/cfergeau/win32-background-launcher
+module github.com/crc-org/win32-background-launcher
```
I'd name this `github.com/crc-org/win32-background-launcher/tools` but I don't know if this is important!

anjannath replied:
updated in the original commit, this should've been part of it so fixed that now in https://github.com/crc-org/win32-background-launcher/pull/1/commits/3be44d1ac35d7410dddb702cf55b31d8479cd8f8

cfergeau reviewed (commented):
One comment left about module name in tools/go.mod. Apart from this, looks good to me/ack!

anjannath committed:
rpm: exclude building on pp64le

GOOS/GOARCH pair windows/ppc64le is not supported

while trying to build it shows the following error:
go: unsupported GOOS/GOARCH pair windows/ppc64le
